include/aws/cal/private/opensslcrypto_common.h

```diff
@@ -23,6 +28,16 @@ struct openssl_hmac_ctx_table {
     hmac_ctx_update update_fn;
     hmac_ctx_final final_fn;
     hmac_ctx_reset reset_fn;
+
+    /* There is slight variance between the crypto interfaces.
+        Note that function pointer casting is undefined behavior.
+        To workaround the issue, use generic pointer for crypto and let delegate
+        function cast it back to correct type.
+        Do not use following fields manually. */
+    struct {
+        crypto_generic_fn_ptr reset_fn;
```

Comment:
I don't actually see the reset_fn() ever getting called from anywhere? Are we actually using it? If not, can we remove all the reset_fn related code?

Reply:
you are right, looks like it was used at some point, but has since been removed. removing it alltogether